### ISSUE TITLE
Rule proposal: `no-document-cookie`

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/eslint-plugin-unicorn#695]
It's not recommended to use `document.cookie` directly as it's easy to get the string wrong. See https://github.com/sindresorhus/eslint-plugin-unicorn/issues/299 and https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie Instead, the user should use a cookie library.

I guess we should only prevent setting `document.cookie`? There might be valid reasons to read it?

**Comment by fisker** [comment]
Have to use `eslint-disable` on this

```js
document.cookie = reallyGoodCookieStringConcatMethod(..);
```

?

**Comment by sindresorhus** [comment]
I guess we can only catch string literals or references to string literals.

**Comment by fisker** [comment]
We'll miss too many cases, like

` 'name=value; expires=' + tomorrow `

Or

`` `name=value; expires=${tomorrow}` ``

**Comment by fisker** [comment]
Let's ignore my case, not many cookie lib returns a string  ask  you  to  assign yourself.

**Comment by sindresorhus** [comment]
Good points. I agree.

**Comment by sindresorhus** [comment]
This is now accepted.

**Comment by fisker** [comment]
Saw a related plugin https://github.com/sota1235/eslint-plugin-no-document-cookie
